### ISSUE TITLE
Taking over domain registrations for AAS WorldWide Telescope project

### ISSUE THREAD

**Issue by pkgw** [dotnet-foundation/projects#2]
Hi,

I'm the leader of the AAS WorldWide Telescope project. We have several domains registered to the project by individual contributors, and I'd like to work on centralizing their management under the DNF. How should I go about starting that process?

While I'm here — I'd also like to review the DNS configuration for the main domain, worldwidetelescope.org. There are some updates to be made but I don't know the full set of subdomains that we actually have defined at the moment.

**Comment by haacked** [comment]
Hi @pkgw, sorry for the delayed response. The board is still getting its house in order.

I admit I'm a little ignorant of how DNS is handled for member projects. So is the main DNS for AAS WorldWide Telescope project under the .NET Foundation's control?

@jongalloway who is our DNS provider? Who maintains that? Perhaps members of the board can pitch in to help get DNS transfers underway.

**Comment by pkgw** [comment]
@haacked To the best that I've been able to determine, yes, the Foundation controls our DNS. I haven't been able to discover a way to modify our DNS config directly or even just view it (except for queries, of course).

FWIW, I was not involved in the project when it came on board with the Foundation, so it is very possible that there is some institutional knowledge that's been lost.

**Comment by haacked** [comment]
Ok, I'll try and follow up with @jongalloway to learn more about how this works. Having the foundation manage the DNS seems not very scalable. I see the benefit in that your project doesn't have to pay for it, so we should see if we can come up with a better approach.

**Comment by jongalloway** [comment]
It looks like worldwidetelescope.org was never transferred from Microsoft to .NET Foundation ([https://whois.icann.org/en/lookup?name=worldwidetelescope.org](whois lookup). That's not a requirement, projects can host their sites however they want, but we probably should transfer it.

For the domains that the .NET Foundation does have registered, we've been using Azure DNS. It's really configurable and easy to share admin access, e.g. both @onovotny and I have been adminning them and can add others on. Or as project leader, if you'd like to transfer the domain to something you directly control, you can do that, too.

Let me know where you'd like to transfer it and I can get that going. I'm partial to Azure DNS since it's worked so well for other projects, but it's up to you.

**Comment by pkgw** [comment]
Errrr, interesting ... with the Microsoft contact name and msft.net DNS servers I had thought that meant that things had been transferred, but I suppose the whole point is that the Foundation is *not* Microsoft.

In which case, I am not sure who actually controls our domain within Microsoft, which is exciting. Do you know anything about this "MarkMonitor" registrar? That might be a clue.

**Comment by jongalloway** [comment]
Transferring from Microsoft won't be a problem, there's a domains support team and I have some PM's I've worked with to get things transferred in the past. Would you like to move to our Azure DNS?

**Comment by jongalloway** [comment]
@pkgw Checking on this - happy to transfer to Azure DNS or somewhere else, let me know.

**Comment by pkgw** [comment]
Thanks for pinging — I lost track of this. I would definitely like to move. What's the risk of downtime?

Also, would the transfer that you're envisioning also involve transferring ownership of the domain, or would that remain a separate item to consider?

**Comment by pkgw** [comment]
Time flies! What are the prospects for pushing this forward?

**Comment by pkgw** [comment]
@jongalloway @onovotny Ping? It's been a while.

Besides the main project domain, one of our core contributors has owned the related domains `openwwt.{org,com}` for a while and would like them to be taken off his hands.

**Comment by jongalloway** [comment]
@ChrisSfanos We'd like to transfer worldwidetelescope.org from Microsoft domains to .NET Foundation account (using our NameCheap account). Can you help us get the domain transfer unlocked from Microsoft domains side?

**Comment by pkgw** [comment]
@ChrisSfanos Would you also have access to our current DNS configuration? I don't actually have this, and would like to make sure that we can get the information set up in Azure before doing any transfers to minimize the chances of wonky downtime.

**Comment by ChrisSfanos** [comment]
Not sure yet - I need some info from @jongalloway so our DNS team can take a look.  I'll ping him again on the thread

**Comment by jongalloway** [comment]
@ChrisSfanos I've just replied on the e-mail thread with Microsoft Domains, including history and supporting documentation.

**Comment by pkgw** [comment]
Hi — checking in on this after the Thanksgiving holiday. Anyone need anything from me? Thanks.

**Comment by pkgw** [comment]
For context, we're looking to make a major release on January 6, and in advance of that we want to switch our web presence to start using Azure Application Gateway, which will require DNS changes that I can't currently make. Is that timescale going to be achievable?

**Comment by jongalloway** [comment]
@pkgw We got the transfer unlocked from the Microsoft side, so we should be able to wrap up the transfer this week. I'll check in with @onovotny and we'll plan out the DNS and registrar changes.

**Comment by pkgw** [comment]
Fun update: in the course of my digging I found another domain for the project: `wwtelescope.org` ... which is associated with MarkMonitor, but its DNS is coming from `azure-dns.org`. Then `wwtelescope.com` seems to be owned by Microsoft and also handled by azure-dns. Fun times!

The priority is definitely `worldwidetelescope.org`, but ideally all of these domains would be grouped into one holding.

**Comment by onovotny** [comment]
Do you have the full list of DNS entries that the domain has so we can set up DNS in the Foundation's azure subscription?

The transfer is in progress though it can take a little while. Transferring it shouldn't change the DNS servers it's currently pointing to.

**Comment by pkgw** [comment]
@onovotny I actually haven't been able to figure out any way to get my hand on the full set of DNS records, which has been one of the frustrating things. Would it be helpful for me to put together a list of the things that I *do* know about?

**Comment by onovotny** [comment]
Sure, any bit helps.

**Comment by pkgw** [comment]
Here's the info from all of the subdomains I can find: [wwt-dig.txt](https://github.com/dotnet-foundation/projects/files/3961439/wwt-dig.txt)

**Comment by onovotny** [comment]
I created a dns zone in your azure subscription (WWT-DNS resource group) and put those entries into it. You should have full control over that resource. Once the transfer is complete, we can update the nameserver records to point to those new ones.

**Comment by jongalloway** [comment]
The transfer is complete. @onovotny - you want to handle the DNS switch? If it's just updating nameservers, I'm happy to do it.

**Comment by onovotny** [comment]
Should just be updating the DNS to point to the new one. @pkgw, can you confirm that you can see/have access to the new DNS Zone in your Azure subscription so you're able to update it directly if desired?

**Comment by pkgw** [comment]
@onovotny Yes, I can see and update the DNS, thanks!

There are a few more domains I mentioned upthread that I'd also like to work on gathering under the Foundation's account. Should I break those into separate issues?

**Comment by onovotny** [comment]
It would probably be easiest to track each domain individually with a separate issue.

**Comment by pkgw** [comment]
It looks like things have fully switched over! Thank you for your help with all this.
